### PR TITLE
gnu-tar: disable SELinux

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

**Comment by dawidd6** [comment]
Why?

**Comment by danielnachun** [comment]
On RHEL/CentOS systems, `gnu-tar` (and many other formulae) will pickup the system `libselinux.so` and try to link to it.  Although the compilation finishes correctly, it results in a non-working executable, because it cannot find `libselinux.so` in the library search path.  Unless we plan to add `libselinux` as a formula, this should be disabled so that it works for RHEL/CentOS users. 

We've already done this for a few other formula (search `selinux` for other examples in this repo).  However if we need to revisit this and either add `libselinux` as a formula or instruct RHEL/CentOS users to symlink their system `libselinux.so` into `$(brew --prefix)/lib`, I'm open to other ideas!

**Comment by iMichka** [comment]
> configure: error: you should not run configure as root (set FORCE_UNSAFE_CONFIGURE=1 in environment to bypass this check)

Can you add that flag (FORCE_UNSAFE_CONFIGURE) to the env for Linux?

File: Formula/gnu-tar.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def install
     ENV["gl_cv_func_getcwd_abort_bug"] = "no" if MacOS.version == :el_capitan
 
     # Fix configure: error: you should not run configure as root
-    ENV["FORCE_UNSAFE_CONFIGURE"] = "1" unless ENV["CI"]
```
Comment (iMichka) [inline Formula/gnu-tar.rb:40]: Ha I see. This should have been:

`ENV["FORCE_UNSAFE_CONFIGURE"] = "1" unless OS.mac? && ENV["CI"]`

Comment (danielnachun) [reply Formula/gnu-tar.rb:40]: Haha, I was really confused as to why that was being disabled in CI when we clearly needed it.  Now it makes sense!

**Review by iMichka — APPROVED** [review]

**Comment by iMichka** [comment]
Thanks @danielnachun !

**Pull request merged**